### PR TITLE
screenshots: update documentation

### LINKED ISSUE #9413: Unclear documentation about screenshot discovery feature
### Describe the issue

Hi,

I'm excited about the new feature for discovering automatically screenshots in a repo in 4.18 (thanks for that!) but the doc about it (in https://docs.weblate.org/en/latest/admin/projects.html#screenshot-file-mask) is not quite clear and we have a disagreement when talking about it.

So I just want to confirm: this is configured per component and the mask aims at matching a translation key, right? 
For example if I have a component `foo` with a translation key `my.translation.key` and a mask  `/foo/translations/screenshots/*.png` then it will search for `/foo/translations/screenshots/my.translation.key.png`. Is that correct? 

I'll improve the doc with an example if it's right.

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

Read the documentation and discuss it.

### Expected behavior

_No response_

### Screenshots

_No response_

### Exception traceback

_No response_

### How do you run Weblate?

PyPI module

### Weblate versions

_No response_

### Weblate deploy checks

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ParthS007**
## Proposed changes

closes #9413 

## Checklist

- [x] Lint and unit tests pass locally with my changes.
- [x] I have added documentation to describe my feature.
- [x] I have squashed my commits into logic units.
- [x] I have described the changes in the commit messages.

### CODE REVIEW

File: docs/admin/projects.rst [hunk 1]
```diff
@@ -437,12 +437,28 @@ to be escaped as ``[[]`` or ``[]]``.
...
+
+- repository_root
+   - component_A
+      - docs
+         - image1.png
+         - image2.jpg
+
+For component_A, you want to allow discovery and updates of PNG screenshots.
+You'd set the screenshot mask for component_A as ``component_A/docs/*.png``.
+This means any PNG images under docs in component_A can be discovered and updated.
```
Comment (surli) [inline docs/admin/projects.rst:459]: So image names are never linked to translation keys AFAIU: it's a manual job to link a discovered screenshot in a component to a specific translation key? Might worth indicating it if it's the case.

Comment (ParthS007) [reply docs/admin/projects.rst:459]: @surli Thanks for your comment. When a new screenshot is discovered in the VCS, a new screenshot object is created and linked to `component.translation` as before, no changes in the behavior there.

Please let me know if that answers your question.

Comment (surli) [reply docs/admin/projects.rst:459]: Partly yes: you confirm that there's no automatic attachment of the screenshot to a particular translation. It remains a manual work to perform the link?

Comment (ParthS007) [reply docs/admin/projects.rst:459]: Yes.

Comment (ParthS007) [reply docs/admin/projects.rst:462]: @surli I have updated the commit. Please let me know if it is clear now.

**Comment by nijel** [comment]
Rendered docs: https://weblate--9434.org.readthedocs.build/en/9434/admin/projects.html#screenshot-file-mask

I wonder if we can improve the repository structure look, the nested lists look a bit confusing here.

File: docs/admin/projects.rst [hunk 2]
```diff
@@ -437,12 +437,30 @@ to be escaped as ``[[]`` or ``[]]``.
...
+2. It is a manual job to link a discovered screenshot in a component to a specific translation key.
+
+For example:
+
+Let's assume your VCS repository has a structure like this:
+
+* repository_root/component_A/docs
+
+  * image1.png
+  * image2.jpg
```
Comment (nijel) [inline docs/admin/projects.rst:457]: This is what we use at https://weblate--9434.org.readthedocs.build/en/9434/admin/customize.html#creating-a-python-module:

```suggestion
.. code-block:: text

    component_A
    └── docs
        ├── image1.png
        └── image2.jpg

```

**Review by orangesunny — APPROVED** [review]

**Pull request merged**